analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodsShouldNotHaveIdenticalImplementations.CSharp9.cs

```diff
@@ -88,3 +88,22 @@ interface SomeInterface
         Console.WriteLine("Result: {0}", s);
     }
 }
+
+public static class TypeConstraints
+{
+    public static int Use<T>(T? value) where T : struct => 1;
+
+    public static int Use<T>(T? value) where T : class => 2;
```

Comment:
This is probably not limited to struct/class, but also every other kind of constraint (e.g. `IComparable`/`IEquatable<T>`). Can you add a test case for such a combination as well?

Reply:
This demonstrates the problem. When working on the issue we can explore different scenarios as well.